pangea-sdk/service/vault/api.go

```diff
@@ -0,0 +1,397 @@
+package vault
+
+import (
+	"context"
+	"errors"
+
+	"github.com/pangeacyber/pangea-go/pangea-sdk/pangea"
+)
+
+func (v *vault) Revoke(ctx context.Context, input *RevokeRequest) (*pangea.PangeaResponse[RevokeResult], error) {
+	if input == nil {
+		return nil, errors.New("nil pointer to struct")
+	}
+
+	req, err := v.Client.NewRequest("POST", "v1/revoke", input)
+	if err != nil {
+		return nil, err
+	}
+	out := RevokeResult{}
+	resp, err := v.Client.Do(ctx, req, &out)
+
+	if resp == nil {
+		return nil, err
+	}
+
+	panresp := pangea.PangeaResponse[RevokeResult]{
+		Response: *resp,
+		Result:   &out,
+	}
+
+	return &panresp, err
+}
+
+func (v *vault) Delete(ctx context.Context, input *DeleteRequest) (*pangea.PangeaResponse[DeleteResult], error) {
+	if input == nil {
+		return nil, errors.New("nil pointer to struct")
+	}
+
+	req, err := v.Client.NewRequest("POST", "v1/delete", input)
+	if err != nil {
+		return nil, err
+	}
+	out := DeleteResult{}
+	resp, err := v.Client.Do(ctx, req, &out)
+
+	if resp == nil {
+		return nil, err
+	}
+
+	panresp := pangea.PangeaResponse[DeleteResult]{
+		Response: *resp,
+		Result:   &out,
+	}
+
+	return &panresp, err
+}
+
+func (v *vault) Get(ctx context.Context, input *GetRequest) (*pangea.PangeaResponse[GetResult], error) {
+	if input == nil {
+		return nil, errors.New("nil pointer to struct")
+	}
+
+	req, err := v.Client.NewRequest("POST", "v1/get", input)
+	if err != nil {
+		return nil, err
+	}
+	out := GetResult{}
+	resp, err := v.Client.Do(ctx, req, &out)
+
+	if resp == nil {
+		return nil, err
+	}
+
+	panresp := pangea.PangeaResponse[GetResult]{
+		Response: *resp,
+		Result:   &out,
+	}
+
+	return &panresp, err
+}
+
+func (v *vault) List(ctx context.Context, input *ListRequest) (*pangea.PangeaResponse[ListResult], error) {
```

Comment:
Do you think we should validate the input here? Order, OrderBy, Filters, etc...

Reply:
No, we shouldn't